test/with_dummyserver/test_connectionpool.py

```diff
@@ -1061,6 +1019,8 @@ def test_headers_not_modified_by_request(
             else:
                 conn = pool._get_conn()
                 conn.request("GET", "/headers", chunked=chunked)
+                conn.getresponse().close()
+                conn.close()
```

Comment:
I was getting ResourceWarnings without closing those properly. Possibly an advantage of using Trio.